internal/vcs/auth.go

```diff
@@ -16,20 +16,23 @@ type GitAuthConfig struct {
 
 // Convert converts HTTPClientConfig to the native Prometheus type. If h is
 // nil, the default client config is returned.
-func (h *GitAuthConfig) Convert() transport.AuthMethod {
+func (h *GitAuthConfig) Convert() (transport.AuthMethod, error) {
 	if h == nil {
-		return nil
+		return nil, nil
 	}
 
 	if h.BasicAuth != nil {
-		return h.BasicAuth.Convert()
+		return h.BasicAuth.Convert(), nil
 	}
 
 	if h.SSHKey != nil {
-		key, _ := h.SSHKey.Convert()
-		return key
+		key, err := h.SSHKey.Convert()
```

Comment:
can be simplified by having directly `return h.SSHKey.Convert()` here

Reply:
Done in HEAD :D